vcs/src/main/java/org/openjdk/skara/vcs/hg/HgRepository.java

```diff
@@ -1494,4 +1494,11 @@ public boolean cherryPick(Hash hash) throws IOException {
             return p.await().status() == 0;
         }
     }
+
+    @Override
+    public int commitCount() throws IOException {
+        try (var p = capture("hg", "id", "--num", "--rev", "tip")) {
+            return Integer.parseInt(await(p).stdout().get(0)) + 1;
```

Comment:
Yes, in Mercurial every commit is assigned an ever increasing numerical ID. The "tip" is defined as the commit with the highest ID number. The first commit has ID 0, so to get the total count we add one to the ID number of "tip". (There are situations where this will not be strictly correct, but counting like this is good enough for this purpose)

Reply:
Thanks for the nice explanation.